code/modules/reagents/chemistry/reagents/toxin_reagents.dm

```diff
@@ -1146,3 +1146,320 @@
+	description = "A revolting mixture of toxic byproducts left over from the production of poisons"
+	color = "#A20067"
+	toxpwr = 1
+	metabolization_rate = 1.5 * REAGENTS_METABOLISM
+
+/datum/reagent/toxin/goop/on_mob_life(mob/living/M)
+	if(prob(15))
+		M.reagents.add_reagent(get_random_toxic_reagent_id(), 1)
+	..()
+
+/datum/reagent/toxin/gibemitter
+	name = "Unstable gibs"
+	description = "This does not smell very nice."
+	color = "#FF1111"
+	toxpwr = 0
+
+/datum/reagent/toxin/gibemitter/on_mob_life(mob/living/carbon/M)
+	if(istype(M))
+		if(prob(20))
+			new /obj/effect/gibspawner/generic(get_turf(holder.my_atom),null,M.dna)
+	..()
+
+/datum/reagent/toxin/bear
+	name = "Bearium"
+	description = "If you like puns and gibbed monkeys you will like this."
+	color = "#CAD15A"
+	toxpwr = 0
+
+/datum/reagent/toxin/bear/on_mob_life(mob/living/carbon/M)
+	if(istype(M))
+		if(prob(25))
+			M.visible_message("<span class='danger'>[M.name] growls strangely.</span>")
+		switch(current_cycle)
+			if(5 to 15)
+				if(prob(15))
+					M.emote(pick("twitch","blink_r","scream"))
+					M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 2)
+			if(15 to 40)
+				if(prob(15))
+					M.vomit(20)
+			if(40 to 60)
+				if(prob(10))
+					M.vomit(20, 0, 4)
+					M.adjustBruteLoss(10)
+					M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10)//this gets much more serious
+			if(60 to INFINITY)
+				if(prob(50))
+					M.visible_message("<span class='danger'>[M.name] explodes in a shower of gibs leaving a space bear!</span>")
+					new /mob/living/simple_animal/hostile/bear(M.loc)
+					M.gib()
+					return ..()
+	..()
+
+/datum/reagent/toxin/methphos
+	name = "Methylphosphonyl difluoride"
+	description = "Maybe you could make something really really toxic out of this?"
+	color = "#C8A5DC"
+	toxpwr = 0.5
+
+/datum/reagent/toxin/sarin_a
+	name = "Translucent mixture"
+	description = "This mixture has a very light white hint to it but is filled with impurities"
+	color = "#AAAACB"
+	toxpwr = 1
+
+/datum/reagent/toxin/sarin_a/on_mob_life(mob/living/M)
+	M.eye_blurry = max(M.eye_blurry, 1)
+	..()
+
+/datum/reagent/toxin/sarin_b
+	name = "diluted sarin"
+	description = "A very impure form of sarin"
+	color = "#CCCCCC"
+
+/datum/reagent/toxin/sarin_b/on_mob_life(mob/living/M)
+	M.eye_blurry = max(M.eye_blurry, 3)
+	if(prob(15))
+		if(iscarbon(M))
+			var/mob/living/carbon/C = M
+			C.vomit(20)
+
+/datum/reagent/toxin/sarin //causes toxin damage, respiratory failure, blurs eyes and drowsiness, extremely lethal unless countered with atropine
+	name = "Sarin"
+	description = "A family friendly lethal nerve agent, handle with care!"
+	color = "#FFFFFF"
+	toxpwr = 0
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	taste_description = "impending death"
+
+/datum/reagent/toxin/sarin/on_mob_life(mob/living/M)
+	M.Jitter(50)
+	M.apply_effect(EFFECT_STUTTER, 5)
+	if(current_cycle % 10 == 0)
+		if(iscarbon(M))
+			var/mob/living/carbon/C = M
+			pick(C.vomit(20, stun = FALSE), C.vomit(20,1, stun = FALSE))
```

Comment:
what the fuck is this, also this is going to call both vomit procs. also use TRUE instead of 1.